synapse/config/oidc_config.py

```diff
@@ -43,6 +44,17 @@ def read_config(self, config, **kwargs):
         except DependencyException as e:
             raise ConfigError(e.message) from e
 
+        # check we don't have any duplicate idp_ids
+        # XXX: this won't detect clashes with other IdP providers using other SSO
+        # mechanisms (such as SAML or CAS); that will be detected when we set up the
+        # listeners but by then synapse will have forked, so it's not ideal.
+        c = Counter([i.idp_id for i in self.oidc_providers])
+        for idp_id, count in c.items():
+            if count > 1:
+                raise ConfigError(
+                    "Multiple OIDC providers have the idp_id %r." % idp_id
+                )
```

Comment:
Should this give a path of `oidc_providers` at least? Not sure if that's even reasonable.

Reply:
maybe, but it could be a clash between an entry in `oidc_providers` and `oidc_config`

Reply:
Ah, right, the backwards compatibility bit. I think this is OK then.